data/ignition/files/usr/local/bin/start-cluster-installation.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+set -e
+
+source common.sh
+
+wait_for_assisted_service
+
+BASE_URL="http://{{.NodeZeroIP}}:8090/api/assisted-install/v2"
+
+# Get cluster id
+cluster_id=$(curl -s -S "${BASE_URL}/clusters" | jq -r .[].id)
+# Get infra_env_id
+infra_env_id=$(curl -s -S "${BASE_URL}/infra-envs"| jq -r .[].id)
+
+known_hosts=0
+
+while [[ "${cluster_status}" != "ready" && "${known_hosts}" != "${host_count}" ]]
```

Comment:
Since known_hosts starts with 0, I think it is possible for this loop to not do what you expect. It is possible for host_count to also be 0 when none of the hosts have registered with assisted-service.

Reply:
Yes but at that time the cluster status will also not be ready. I agree that reading the `controlPlaneAgents` from the `provisionRequirements` of `AgentClusterInstall` is a much better option to know the exact number of hosts we should be checking against.

Reply:
Done